integration/kubernetes/vfio.sh

```diff
@@ -0,0 +1,181 @@
+			fi
+		fi
+	fi
+
+	# enable debug
+	sudo sed -i -e 's/^#\(enable_debug\).*=.*$/\1 = true/g' \
+		-e 's/^kernel_params = "\(.*\)"/kernel_params = "\1 agent.log=debug"/g' \
+		"${SYSCONFIG_FILE}"
+}
+
+run_test() {
+	local image_type="${1:-}"
+	[ -n "$image_type" ] || die "need image type"
+
+	local machine_type="${2:-}"
+
+	local hypervisor="${3:-}"
+	[ -n "$hypervisor" ] || die "need hypervisor"
+
+	local sandbox_cgroup_only="${4:-}"
+	[ -n "$sandbox_cgroup_only" ] || die "need sandbox cgroup only"
+
+	setup_configuration_file "$image_type" "$machine_type" "$hypervisor" "$sandbox_cgroup_only"
+
+	sudo -E kubectl apply -f "${SCRIPT_DIR}/runtimeclass_workloads/vfio.yaml"
+
+	pod_name=vfio
+	sudo -E kubectl wait --for=condition=Ready pod "${pod_name}"
+
+	# Expecting 2 network interaces -> 2 mac addresses
+	mac_addrs=$(sudo -E kubectl exec -ti "${pod_name}" ip a | grep "link/ether" | wc -l)
+	if [ ${mac_addrs} -ne 2 ]; then
+		die "Error: expecting 2 network interfaces, Got: $(kubectl exec -ti "${pod_name}" ip a)"
+	else
+		info "Success: found 2 network interfaces"
+	fi
+}
+
+main() {
+	# Init k8s cluster
+	${SCRIPT_DIR}/init.sh
+
+	sudo modprobe vfio
+	sudo modprobe vfio-pci
+
+	# unbind device from driver
+	# PCI address
+	local addr="00:03.0"
+	echo 0000:${addr} | sudo tee /sys/bus/pci/devices/0000:${addr}/driver/unbind
+
+	# Create a new VFIO device
+	# Ethernet controller: Red Hat, Inc. Virtio network device
+	# vendor ID: 1af4
+	# device ID: 1041
+	local vendor_id="1af4"
+	local device_id="1041"
+	echo "${vendor_id} ${device_id}" | sudo tee /sys/bus/pci/drivers/vfio-pci/new_id
+
+	# Install network (device) plugin
+	sriov_plugin_url=$(get_version "plugins.sriov-network-device.url")
+	sriov_plugin_version=$(get_version "plugins.sriov-network-device.version")
+	git clone --depth=1 "${sriov_plugin_url}"
+	pushd sriov-network-device-plugin
+	git checkout "${sriov_plugin_version}"
+	sed -i 's|resourceList.*|resourceList": [{"resourceName":"virtio_net","selectors":{"vendors":["'"${vendor_id}"'"],"devices":["'"${device_id}"'"],"drivers":["vfio-pci"],"pfNames":["eth1"]}},{|g' deployments/configMap.yaml
+	sudo -E kubectl create -f deployments/configMap.yaml
+	sudo -E kubectl create -f deployments/k8s-v1.16/sriovdp-daemonset.yaml
+	sleep 5
+	popd
+	rm -rf sriov-network-device-plugin
+
+	sriov_pod="$(sudo -E kubectl --namespace=kube-system get pods --output=name | grep sriov-device-plugin | cut -d/ -f2)"
+	sudo -E kubectl --namespace=kube-system wait --for=condition=Ready pod "${sriov_pod}"
+
+	# wait for the virtio_net resource
+	for _ in $(seq 1 30); do
+		v="$(sudo -E kubectl get node $(hostname) -o json | jq '.status.allocatable["intel.com/virtio_net"]')"
+		[ "${v}" == \"1\" ] && break
+		sleep 5
+	done
+
+	run_test image "" cloud-hypervisor false
+	run_test image "" cloud-hypervisor true
+	# run_test initrd "" cloud-hypervisor false
+	# run_test initrd "" cloud-hypervisor false
```

Comment:
?

Reply:
a request from @likebreath

Reply:
I have filed an issue to tack this, thanks